src/main.rs

```diff
@@ -46,6 +50,33 @@ fn run() -> FeludaResult<()> {
         &format!("Starting Feluda with args: {:?}", args),
     );
 
+    // Handle repository cloning if --repo is provided
+    let (analysis_path, _temp_dir) = match args.repo {
+        Some(repo_url) => {
+            log(LogLevel::Info, &format!("Attempting to clone repository: {}", repo_url));
+            let temp_dir = TempDir::new().map_err(|e| {
+                FeludaError::Unknown(format!("Failed to create temporary directory: {}", e))
+
+            })?;
+            let repo_path = temp_dir.path();
+
+            // Clone the repository
+            if let Err(e) = clone_repository(&repo_url, repo_path, args.token.as_deref(), args.ssh_key.as_deref(), args.ssh_passphrase.as_deref()) {
```

Comment:
It's better to pass `args` together.
That way, in future, if there's more or optional, won't have to change the signature here.